kernel/src/arch/x86/cpu.rs

```diff
@@ -167,6 +171,20 @@ impl TryFrom<&CpuException> for PageFaultInfo {
     }
 }
 
+enum CpuVendor {
+    Intel,
+    AMD,
```

Comment:
```suggestion
    Amd,
```
We should prefer `Amd` instead of `AMD`. In the Rust standard library, we use [`TcpSocket`](https://doc.rust-lang.org/std/net/struct.TcpListener.html) instead of `TCPSocket`.